### ISSUE TITLE
Won't build from 2.10 to 2.13

### ISSUE THREAD

**Issue by giann** [at-internet/atinternet-apple-sdk#53]
We're migrating our app to Swift 4.2.

Here is the full xcode log:

```
/.../Pods/ATInternet-Apple-SDK/ATInternetTracker/Sources/Dispatcher.swift:119:23: error: use of unresolved identifier 'Crash'
        let report = (Crash.compute() as NSDictionary?) as! [String: Any]?
                      ^~~~~
ObjectiveC.NSObject:102:14: note: did you mean 'hash'?
    open var hash: Int { get }
             ^
ObjectiveC.NSObject:4:17: note: did you mean 'hash'?
    public func hash(into hasher: inout Hasher)
                ^
CoreFoundation.CFHash:1:13: note: did you mean 'CFHash'?
public func CFHash(_ cf: CFTypeRef!) -> CFHashCode
            ^
/.../Pods/ATInternet-Apple-SDK/ATInternetTracker/Sources/Screen.swift:212:9: error: use of unresolved identifier 'Crash'
        Crash.lastScreen(builtScreenName)
        ^~~~~
ObjectiveC.NSObject:102:14: note: did you mean 'hash'?
    open var hash: Int { get }
             ^
ObjectiveC.NSObject:4:17: note: did you mean 'hash'?
    public func hash(into hasher: inout Hasher)
                ^
CoreFoundation.CFHash:1:13: note: did you mean 'CFHash'?
public func CFHash(_ cf: CFTypeRef!) -> CFHashCode
            ^
/.../Pods/ATInternet-Apple-SDK/ATInternetTracker/Sources/StringExtension.swift:53:16: error: use of unresolved identifier 'Hash'; did you mean 'hash'?
        return Hash.sha256Value("AT" + self)
               ^~~~
               hash
Swift.String:2:17: note: 'hash' declared here
    public func hash(into hasher: inout Hasher)
                ^
Swift.StringProtocol:2:17: note: 'hash' declared here
    public func hash(into hasher: inout Hasher)
                ^
Foundation.StringProtocol:20:16: note: 'hash' declared here
    public var hash: Int { get }
               ^
/.../Pods/ATInternet-Apple-SDK/ATInternetTracker/Sources/Tracker.swift:1512:22: error: use of unresolved identifier 'Crash'
        if let map = Crash.recover() as? [String : Any] {
                     ^~~~~
ObjectiveC.NSObject:102:14: note: did you mean 'hash'?
    open var hash: Int { get }
             ^
ObjectiveC.NSObject:4:17: note: did you mean 'hash'?
    public func hash(into hasher: inout Hasher)
                ^
CoreFoundation.CFHash:1:13: note: did you mean 'CFHash'?
public func CFHash(_ cf: CFTypeRef!) -> CFHashCode
            ^
/.../Pods/ATInternet-Apple-SDK/ATInternetTracker/Sources/Tracker.swift:1595:21: error: use of unresolved identifier 'Crash'
                    Crash.handle()
                    ^~~~~
ObjectiveC.NSObject:95:21: note: did you mean 'hash'?
    open class func hash() -> Int
                    ^
ObjectiveC.NSObject:4:17: note: did you mean 'hash'?
    public func hash(into hasher: inout Hasher)
                ^
CoreFoundation.CFHash:1:13: note: did you mean 'CFHash'?
public func CFHash(_ cf: CFTypeRef!) -> CFHashCode

```

**Comment by d-theo** [comment]
Its looks like xCode can't import the C headers of the library.
Can you share your Podfile and your Project Settings, like Other Link Flags or other C related stuff ?

**Comment by giann** [comment]
Sorry for the late response.
Can you give an exhaustive list of the settings you'd like me to share?
I can't really dump all of it here publicly.

**Comment by giann** [comment]
Btw, I tried 2.14.0 without success either.

**Comment by AmineZ** [comment]
I have the same problem, i added `Library Search Paths = $(TOOLCHAIN_DIR)/usr/lib/swift/$(PLATFORM_NAME)` in build settings but the app crash again.

**Comment by mackoj** [comment]
@d-theo Hi, we have this issue too. @AmineZ set it back to the 2.8.9 version then upgrade it for Swift 5 and now it is working properly. Please consider fix it.

**Comment by d-theo** [comment]
It's kinda hard to reproduce the issue but we are working on it :)

**Comment by nsagnett** [comment]
Hi @mackoj , it's too hard to reproduce this issue, can U give us some more informations or submit a PR to fix it ?

**Comment by mackoj** [comment]
hi @nsagnett our fix consist on using an old version of `atinternet-apple-sdk/Tracker`, migrate it manually to Swift 5 and add `Library Search Paths = $(TOOLCHAIN_DIR)/usr/lib/swift/$(PLATFORM_NAME)` to the Podspec. I'm on vacation @AmineZ will come back at you with a more detailed explanation.

**Comment by AmineZ** [comment]
Hi @nsagnett, i can't help you to unfortunately.

Since we migrate our projet to Swift 5, the application doesn't crash when it launched from Xcode directly. But when we disconnect the device from Xcode and launch a seconde time the application crash immediately on start.

From the crash log and some help in internet, the problem is caused by a problem in a dependency in our project.

Deleting `AT Internet / Tracker` from our projet resolve the problem. 
So we rollback to an old AT Internet version  `2.8.9` and i migrate it to Swift 5 to compile.

I tried to rollback AT Internet to `2.9.0` version but projet crash again.

I also tried to create a new empty projet and import AT Internet using cocoa-pods as we do on our projet. The result is that the application doesn't crash 😩

So why i tell you that i can't help you. Something is wrong when you migrate AT-Internet to swift 5 maybe that does not work on our project.

For information, our projet is a mix of Objective-C and Swift code (60% swift).
Actually we use `2.8.9` version and we can't update it.
